### PR TITLE
Fix Input Object Validation Failing with Invalid Values

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#2003](https://github.com/ballerina-platform/ballerina-standard-library/issues/2003)

## Examples
N/A

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [[#1984] Fix Not Allowing to Pass Default Value to Enum Type Variables](https://github.com/ballerina-platform/ballerina-standard-library/issues/1984)
 - [[#1990] Fix Accepting Enum Value as Default Value in String Type Variables](https://github.com/ballerina-platform/ballerina-standard-library/issues/1990)
 - [[#1998] Fix Ignoring Parameters after Context Parameter](https://github.com/ballerina-platform/ballerina-standard-library/issues/1998)
+- [[#2003] Fix Failing Input Object Validation with Invalid Values](https://github.com/ballerina-platform/ballerina-standard-library/issues/2003)
```
Comment (ThisaruGuruge) [inline changelog.md:38]: This double negative doesn't seems right. (Shouldn't input object validation fail with invalid values?).

File: ballerina/tests/resources/expected_results/input_variables_with_empty_input_object_value.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "errors": [
+        {
+            "message": "Variable \"$userName\" got invalid value {};Expected type String!. String cannot represent value: {}",
```
Comment (ThisaruGuruge) [inline ballerina/tests/resources/expected_results/input_variables_with_empty_input_object_value.json:4]: Shouldn't we have a space here?

```suggestion
            "message": "Variable \"$userName\" got invalid value {}; Expected type String!. String cannot represent value: {}",
```

File: ballerina/validator_visitor.bal [hunk 3]
```diff
@@ -123,23 +123,30 @@ class ValidatorVisitor {
...
+                    if ((subInputValue.'type).kind == NON_NULL && schemaArg?.defaultValue is ()) {
+                        string message = string`Field "${fieldName}" argument "${subInputValue.name}" of type ` +
+                        string`"${getTypeNameFromType(subInputValue.'type)}" is required, but it was not provided.`;
+                        self.errors.push(getErrorDetailRecord(message, argumentNode.getLocation()));
+                    }
                 }
             }
+        } else {
+            string expectedTypeName = getOfType(schemaArg.'type).name.toString();
+            string message = string`${expectedTypeName} cannot represent non ${expectedTypeName} value: {}`;
```
Comment (ThisaruGuruge) [inline ballerina/validator_visitor.bal:147]: Is this the correct way, or shall we use `null` here instead of `{}`?

Comment (DimuthuMadushan) [reply ballerina/validator_visitor.bal:147]: GraphQL playground is also used `{}` in error messages. shall we keep it as same? WDYT?

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**